### ISSUE TITLE
Single stream can miss events

### ISSUE THREAD

**Issue by stevenmyhre** [prwnr/laravel-streamer#54]
When configuring a Streamer to just listen to one stream name, the `awaitSingle` method will use a new entries key of `$`.

This has the potential to miss events when there is a `stream_read_timeout` set.

Example chain of events

```
XADD {a}
XREAD BLOCK {timeout} $
-- timeout hit
XADD {b}
XREAD BLOCK {timeout} $
```

The consumer would end up missing the second XADD's data, since it is only asking for new items from the time it executed XREAD.

I wrote my own solution to get past this (using the last message ID as the id for the XREAD), but thought others should know in case they run into this strange behavior.

**Comment by prwnr** [comment]
@stevenmyhre would you be willing to contribute your change for this behavior?

**Comment by stevenmyhre** [comment]
The solution I made did not fit into this package nicely, otherwise I would have provided a pull request. I coded from scratch.

**Comment by prwnr** [comment]
@abublihi I am sorry, but I didn't have the time to look into this, I hoped that OP will open a PR with his changes and we could come up with suitable solution. 

If this impacts you as well, I will try to devote some time to see if I can improve this part.

However, if I am not mistaken, the fastest solution for this problem is to use group+consumer arguments - this will switch the listening to grouped one, which uses `>` instead of `$` - it will result in no messages lost (because Redis in this mode will be always popping unreceived - for this particular group - messages. Bear in mind that starting listening on the stream with different group will send out ALL the messages to it - because it acknowledges message being received only for one group). This approach is the one that my company uses in production and we did not face problems with lost messages (at least to the ones we cared about).

**Comment by prwnr** [comment]
Given that here https://github.com/prwnr/laravel-streamer/blob/35fdd6dea8c978c5d875116c08721fbdb808f37e/src/Stream/MultiStream.php#L99 the method accepts "lastSeenId" and that the part that triggers it: https://github.com/prwnr/laravel-streamer/blob/35fdd6dea8c978c5d875116c08721fbdb808f37e/src/EventDispatcher/Streamer.php#L167 gets the ID from the message it receives and uses for the next blocking section, it seems like it should work for the $ option. 

@stevenmyhre but I guess it's a problem when, as you mentioned, timeout hits, so that the payload is empty, and in that case that code fallbacks to "new entry key" which is $.

In that case it perhaps could be solved by changing that part to see if the $lastSeenId is different than getNewEntriesKey and should be used instead. 


I am atm outside of hometown for a while, so I can't verify this myself, but seems a reasonable thing to do and might solve the problem.

**Comment by stevenmyhre** [comment]
Yes I wanted non-grouped, but guaranteed sequential so I implemented a solution with ```>``` and tracking the last ID seen.
